Client/src/pages/suggestionHistory/suggestionHistory.tsx

```diff
@@ -5,97 +5,89 @@ import CalendarTodayIcon from '@mui/icons-material/CalendarToday';
 import SentimentDissatisfiedIcon from '@mui/icons-material/SentimentDissatisfied';
 import SentimentSatisfiedIcon from '@mui/icons-material/SentimentSatisfied';
 import SentimentSatisfiedAltIcon from '@mui/icons-material/SentimentSatisfiedAltOutlined';
+import { Mood } from '@/types/enums/Moods';
+import { MOOD_CATEGORIES, MoodCategory } from '@/types/enums/MoodGroup';
 
-interface Suggestion {
-  title: string;
-  content: string;
-  rank: number;
-  date: Date;
-  eventTitle: string;
+interface FlatEvent {
+  eventId: number;
+  userId: number;
   eventContent: string;
-  eventMoodCategory: MoodCategory;
+  eventDate: string; 
+  mood: Mood;
+  suggestionId: number;
+  suggestionTitle: string;
+  suggestionContent: string;
+  suggestionRank: number; 
+  suggestionExecutionDate: string;
+  // keywords: [];
 }
 
-enum MoodCategory {
-  Positive = "Positive",
-  Negative = "Negative",
-  Regular = "Regular",
-}
-
-const suggestionsList: Suggestion[] = [
-  {
-    eventTitle: "Morning Jog",
-    eventContent: "Went for a 5km jog in the park. Felt great.",
-    title: "Stretching",
-    content: "Do some stretching exercises to cool down.",
-    rank: 2,
-    date: new Date(),
-    eventMoodCategory: MoodCategory.Positive
-  },
-  {
-    eventTitle: "Evening Walk",
-    eventContent: "Had a relaxing walk in the neighborhood.",
-    title: "Hydrate",
-    content: "Drink plenty of water after your walk.",
-    rank: 3,
-    date: new Date(),
-    eventMoodCategory: MoodCategory.Regular
-  },
-  {
-    eventTitle: "Evening Walk",
-    eventContent: "Had a relaxing walk in the neighborhood.",
-    title: "Hydrate",
-    content: "Drink plenty of water after your walk.",
-    rank: 1,
-    date: new Date(),
-    eventMoodCategory: MoodCategory.Negative
-  },
-  {
-    eventTitle: "Evening Walk",
-    eventContent: "Had a relaxing walk in the neighborhood.",
-    title: "Hydrate",
-    content: "Drink plenty of water after your walk.",
-    rank: 3,
-    date: new Date(),
-    eventMoodCategory: MoodCategory.Regular
-  },
-  // Add more suggestions as needed
-];
-
 const HistoryOfSuggestionsPage: React.FC = () => {
-  const [suggestions, setSuggestios] = useState<Suggestion[]>([]);
+  const [suggestions, setSuggestios] = useState<FlatEvent[]>([]);
 
   useEffect(() => {
-    // Todo: get dailySharings of today
-    // const events: Suggestion[] = [];
+      async function fetchMyAPI() {
+        // TODO: Update to SERVER_URL
```

Comment:
use ReactQuery to control the state of the request. for example:
const {
        data: dailySharings,
        isLoading: isDailySharingsLoading,
        isError: isDailySharingsError,
    } = useQuery<DailySharing[]>({
        queryKey: ["events", { date: time }],
        queryFn: async () => (await backendAxiosInstance.get("/events", {
            params: {
                date: time
            }
        })).data,
    });